src/containers/Application.jsx

```diff
@@ -142,22 +147,83 @@ class Application extends Component {
       })
       .catch(e => {
         console.error(e); // eslint-disable-line no-console
+        this.authMetaMask();
         this.setState({ isLoading: false, hasError: false });
       });
 
-    GivethWallet.getCachedKeystore()
-      .then(keystore => {
-        // TODO: change to getWeb3() when implemented. actually remove provider from GivethWallet
-        const provider = this.state.web3 ? this.state.web3.currentProvider : undefined;
-        return GivethWallet.loadWallet(keystore, provider);
-      })
-      .then(wallet => {
-        getWeb3().then(web3 => web3.setWallet(wallet));
-        this.setState({ wallet });
+      const { web3 } = window;
+
+      // check if web3 has been injected into dom window
+      if (typeof web3 !== 'undefined') {
+        const web3js = new Web3(web3.currentProvider);
```

Comment:
everything regarding `web3` should be handled in the `getWeb3` function.

Basically if we detect an injected web3 instance, we pass that provider to our own instance. https://github.com/MetaMask/faq/blob/master/DEVELOPERS.md#partly_sunny-web3---ethereum-browser-environment-check